app/lib/xos.py

```diff
@@ -5,13 +5,22 @@
 XOS_AUTH_TOKEN = os.environ['XOS_AUTH_TOKEN']
 XOS_API_ENDPOINT = os.environ['XOS_API_ENDPOINT']
 
-def update_xos_with_stub_video(video_data):
+def get_or_create_xos_stub_video(video_data):
     xos_video_endpoint = f'{XOS_API_ENDPOINT}assets/'
     headers = {'Authorization': 'Token ' + XOS_AUTH_TOKEN}
+
+    get_response = requests.get(xos_video_endpoint+'?title_contains=NOT%20UPLOADED&checksum='+video_data['master_metadata']['checksum'], headers=headers)
+    get_response.raise_for_status()
+    get_response_json = get_response.json()
+    if get_response_json['count'] == 1:
```

Comment:
It's unlikely but this could return more than 1 result. For example if someone manually created an asset entry in XOS. Just to be safe, could we change this to `> 0`? Otherwise this'll create an additional `Asset` on top of the multiple existing ones.